### PR TITLE
Add neighborhood mini app options to Python Lab levels

### PR DESCRIPTION
**Pull request by molly-moen**
To start creating a version of neighborhood in Python Lab we need the ability to set some Python Lab levels as neighborhood levels. I followed a very similar setup to Java Lab here, with a few tweaks.

In levelbuilder, we now have Python Lab fields (before it was an empty section of the level edit page). There is a "mini app" dropdown, with the option of either "None" or "Neighborhood". In Java Lab we call this CSA View Mode, and since that name did not make sense for Python Lab I went with "mini app" since that is what we have landed on as a name for this feature generally. When neighborhood is selected, we also show fields for start direction and serialized maze. 

If levelbuilders have the start direction and serialized maze filled in, then switch to a non-neighborhood mini app, on save we will clear out those fields. We don't do this in Java Lab but I added that here because it didn't make sense to clutter the level configuration with confusing, unused information. On save we also validate the serialized maze is valid, same as Java Lab. I added unit tests for the serialized maze validation (mostly copied from Java Lab).

I also updated Python Lab to have an override method for `summarize_for_lab2`, so we can set the serialized maze to the template level's maze if the level does not have a serialized maze. I don't think it's likely other lab2 levels will have a serialized maze, so it made sense to keep this logic unique to python lab.

### "None" selected
Screenshot 2024-12-10 at 3 56 52 PM

### "Neighborhood" selected
Screenshot 2024-12-10 at 3 56 44 PM

## Links
- jira ticket: [CT-927](https://codedotorg.atlassian.net/browse/CT-927)

## Testing story
Tested locally

## Follow-up work
Lots of work still to do, see the Intro to Programming release in Codebridge for more tasks!

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/app/models/levels/pythonlab.rb [hunk 1]
```diff
@@ -99,4 +110,45 @@ def get_validations
...
+    level_properties[:serializedMaze] = get_serialized_maze
+    level_properties
+  end
+
+  def get_serialized_maze
+    serialized_maze || project_template_level&.try(:serialized_maze)
+  end
+
+  def clean_up_mini_app_settings
+    if mini_app != 'neighborhood'
```
Comment (fisher-alice) [inline dashboard/app/models/levels/pythonlab.rb:125]: Nice!

File: dashboard/app/models/levels/pythonlab.rb [hunk 2]
```diff
@@ -50,6 +53,14 @@ def self.create_from_level_builder(params, level_params)
     )
   end
 
+  def self.start_directions
+    [['None', nil], ['North', 0], ['East', 1], ['South', 2], ['West', 3]]
```
Comment (fisher-alice) [inline dashboard/app/models/levels/pythonlab.rb:57]: Just curious - if a maze is defined, is there a default direction if 'None' is selected?

File: dashboard/app/models/levels/pythonlab.rb (hunk 1 above)
Comment (fisher-alice) [reply dashboard/app/models/levels/pythonlab.rb:125]: Nit: could add a test for this new feature if not too much trouble

**Review by fisher-alice — APPROVED** [review]
Thanks for the clear PR description! Nit suggestion (you can take or leave).

File: dashboard/app/models/levels/pythonlab.rb (hunk 2 above)
Comment (molly-moen) [reply dashboard/app/models/levels/pythonlab.rb:57]: I believe it defaults to East

File: dashboard/app/models/levels/pythonlab.rb (hunk 1 above)
Comment (molly-moen) [reply dashboard/app/models/levels/pythonlab.rb:125]: added!

**Pull request merged**